### PR TITLE
feat: Add handling for ITensor mean and var in batch_norm

### PR DESCRIPTION
**Pull request by chohk88**
# Description

Support ITensor type running_mean and running_var arguments for Batch Norm converter.

Fixes # ([issue](https://github.com/pytorch/TensorRT/issues/3074))

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py [hunk 1]
```diff
@@ -36,59 +36,136 @@ def batch_norm(
...
+    if isinstance(running_mean, TRTTensor) or isinstance(running_var, TRTTensor):
+        # Default values if weight, bias, running_mean, running_var are None
+        if weight is None:
+            weight = get_trt_tensor(ctx, 1.0, f"{name}_weight", input.dtype)
+        if bias is None:
+            bias = get_trt_tensor(ctx, 0.0, f"{name}_bias", input.dtype)
+        if running_mean is None:
+            running_mean = get_trt_tensor(ctx, 0.0, f"{name}_running_mean", input.dtype)
+        if running_var is None:
+            running_var = get_trt_tensor(ctx, 1.0, f"{name}_running_var", input.dtype)
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py:64]: 1) Do we need to cast these parameters to the type of `input`? I think there's probably a case that `input` is int type while `weight`, `bias`, `running_mean`, and/or `running_var` are float type. It seems problematic to force cast float to int. The `dtype` is optional so you can just leave it blank.

2) `weight` and `bias` could be ITensor as well right?

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py [hunk 2]
```diff
@@ -36,59 +36,136 @@ def batch_norm(
     input: TRTTensor,
     weight: Optional[Union[torch.Tensor, np.ndarray]],
     bias: Optional[Union[torch.Tensor, np.ndarray]],
-    running_mean: Optional[Union[torch.Tensor, np.ndarray]],
-    running_var: Optional[Union[torch.Tensor, np.ndarray]],
+    running_mean: Union[TRTTensor, Optional[Union[torch.Tensor, np.ndarray]]],
+    running_var: Union[TRTTensor, Optional[Union[torch.Tensor, np.ndarray]]],
```
[the remark this answers is not in the document]
Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py:40]: I just noticed that `torch.ops.aten.native_batch_norm.default` and `torch.ops.aten.batch_norm.default` reuse the function. Since they require `running_mean` and `running_var` to be optional, you can put all these types to `Optional[Union[TRTensor, torch.Tensor, np.ndarray]]`

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py (hunk 1 above)
Comment (keehyuna) [reply py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py:64]: >>Do we need to cast these parameters to the type of input
I thinks this is required for [strongly-typed-networks](https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#strongly-typed-networks), different type can be allowed for weak typed network.
I noticed it by enabling strongly typed networks with some model. Here is [some changes](https://github.com/keehyuna/TensorRT/commit/2e32bde7370bf26d8f7b7112b8aa53b52ab39644#diff-47e9eb66cba97ff59bf6c1355e0506921be5780f8796d725f9dc3abcdb22cda5R127) to keep same type in ops. I only saw float32 and half float, I'm now sure if float and int tensor in same layer is possible.

**Review by peri044 — APPROVED** [review]
LGTM

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py (hunk 1 above)
Comment (chohk88) [reply py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py]: ->>Do we need to cast these parameters to the type of input?
If `input_val` is a `TRTTensor`, the input is returned unchanged ([code](https://github.com/pytorch/TensorRT/blob/db15ecaa6f18cfdae99e14b5ff497a60abcf231b/py/torch_tensorrt/dynamo/conversion/converter_utils.py#L386)), so the type stays the same. If `input_val` is something else, `create_constant` converts the type using `to_numpy` ([code](https://github.com/pytorch/TensorRT/blob/db15ecaa6f18cfdae99e14b5ff497a60abcf231b/py/torch_tensorrt/dynamo/conversion/converter_utils.py#L344)). However, if the `value` is an `np.ndarray` or `torch.Tensor`, it keeps the original type ([code](https://github.com/pytorch/TensorRT/blob/db15ecaa6f18cfdae99e14b5ff497a60abcf231b/py/torch_tensorrt/dynamo/conversion/converter_utils.py#L550) and [code](https://github.com/pytorch/TensorRT/blob/db15ecaa6f18cfdae99e14b5ff497a60abcf231b/py/torch_tensorrt/dynamo/conversion/converter_utils.py#L559)). This means setting `input.dtype` has no effect.

I confirmed this behavior when I removed the `input.dtype` argument.

As for the issue @keehyuna  mentioned, it's new to me, so I'll investigate.

->> `weight` and `bias` could be ITensor as well right?
Additionally, according to the schema, `weight` and `bias` can't be `ITensor`, but the converter works fine. I've tested it, and it works successfully.

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (chohk88) [reply py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py]: Thank you for your comments! I've resolved the issue based on your feedback.

**Comment by chohk88** [comment]
> LGTM

If the CI results show no more batch_norm errors, it's good to merge.

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py (hunk 1 above)
Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py]: > ->> weight and bias could be ITensor as well right?
Additionally, according to the schema, weight and bias can't be ITensor, but the converter works fine. I've tested it, and it works successfully.

@chohk88 Did I miss something? The schema is: 
```
- func: batch_norm(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps, bool cudnn_enabled) -> Tensor
- func: _native_batch_norm_legit_no_training(Tensor input, Tensor? weight, Tensor? bias, Tensor running_mean, Tensor running_var, float momentum, float eps) -> (Tensor, Tensor, Tensor)
```

My understanding to `Tensor? weight, Tensor? bias` is that they could be None, ITensor, torch.Tensor, or np.ndarray. Please correct me if I'm wrong.

Comment (chohk88) [reply py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py]: Oh, I misunderstood that. You're right—`weight`, `bias`, `running_mean`, and `running_var` can all be `TRTTensor`. I’ve combined the separate converter and added some test cases for this.

**Comment by chohk88** [comment]
@zewenli98  Although it's beyond the scope of this PR, I noticed something regarding the converters for `layer_norm`, `group_norm`, and `native_group_norm`.

Currently, `weight` and `bias` are defined as:
```python
weight: Optional[Union[torch.Tensor, np.ndarray]],
bias: Optional[Union[torch.Tensor, np.ndarray]],
```

Is this incorrect? Fortunately, unlike batch norm, there's no issue with adding `eps` or applying `to_numpy` on the `weight` or `bias` of `TRTTensor` (or `ITensor`), so it doesn't seem to cause any problems with the converter. However, it seems like the case where the default value for `weight` in `layer_norm` is `None` might not be handled properly.

Here are schema:
- [`group_norm`](https://github.com/pytorch/pytorch/blob/d990dada86a8ad94882b5c23e859b88c0c255bda/aten/src/ATen/native/native_functions.yaml#L2961)
- [`layer_norm`](https://github.com/pytorch/pytorch/blob/d990dada86a8ad94882b5c23e859b88c0c255bda/aten/src/ATen/native/native_functions.yaml#L3250)
- [`native_group_norm`](https://github.com/pytorch/pytorch/blob/d990dada86a8ad94882b5c23e859b88c0c255bda/aten/src/ATen/native/native_functions.yaml#L2963)

**Comment by peri044** [comment]
Merging this as the other failures are unrelated.

**Comment by zewenli98** [comment]
@chohk88 Thanks for pointing out the issue. I think your understanding is correct. Although there's no errors out now, they should be `Optional[Union[TRTTensor, torch.Tensor, np.ndarray]]`. Since the Issue #3114 has been opened above to track it, could you modify them to `Optional[Union[TRTTensor, torch.Tensor, np.ndarray]]` like what you did for the batch_norm?
